### PR TITLE
Add a class that initializes Pinocchio for SOLO 12 without Pybullet

### PR DESCRIPTION
**Pull request by swuuu**
[//]: # "Thanks for your contribution.  To make life of the reviewers easier,"
[//]: # "please give this pull request a meaningful title and provide the"
[//]: # "requested information below."

## Description

[//]: # "Description of what you did.  If it is more complex, consider to add a"
[//]: # "'Summary' section."

I added a class that initializes Pinocchio for SOLO 12 without PyBullet, since we are using Gazebo. 

## How I Tested

[//]: # "Explain how you tested your changes"

I ran the reactive planner that uses this project on Gazebo. The instructions to run the project is [here](https://github.com/swuuu/ros2_controllers_solo12)

[//]: # "Link other open pull request here that need to be merged first."
[//]: # "Remove this section if it does not apply."

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [x] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [x] All new functions/classes are documented and existing documentation is updated according to changes.
- [x] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

File: src/robot_properties_solo/solo12wrapper.py [hunk 1]
```diff
@@ -132,3 +132,65 @@ def reset_to_initial_state(self) -> None:
         q0 = np.matrix(Solo12Config.initial_configuration).T
         dq0 = np.matrix(Solo12Config.initial_velocity).T
         self.reset_state(q0, dq0)
+
+"""
+Similar to the class above, but without PyBullet. Used for ROS + Gazebo projects
+"""
+class Solo12RobotWithoutPybullet():
```
Comment (nim65s) [inline src/robot_properties_solo/solo12wrapper.py]:
```suggestion
class Solo12RobotWithoutPybullet():
    """
    Similar to the class above, but without PyBullet. Used for ROS + Gazebo projects
    """
```

File: src/robot_properties_solo/solo12wrapper.py [hunk 2]
```diff
@@ -132,3 +132,65 @@ def reset_to_initial_state(self) -> None:
...
+        self.joint_names = controlled_joints
+        self.nb_ee = len(self.end_effector_names)
+
+        self.hl_index = self.pin_robot.model.getFrameId("HL_ANKLE")
+        self.hr_index = self.pin_robot.model.getFrameId("HR_ANKLE")
+        self.fl_index = self.pin_robot.model.getFrameId("FL_ANKLE")
+        self.fr_index = self.pin_robot.model.getFrameId("FR_ANKLE")
+
+    def forward_robot(self, q=None, dq=None):
+        if not q:
```
Comment (nim65s) [inline src/robot_properties_solo/solo12wrapper.py]:
```suggestion
        if q is None:
```

File: src/robot_properties_solo/solo12wrapper.py [hunk 3]
```diff
@@ -132,3 +132,65 @@ def reset_to_initial_state(self) -> None:
...
+            q, dq = self.get_state()
+        elif not dq:
```
Comment (nim65s) [inline src/robot_properties_solo/solo12wrapper.py]:
```suggestion
        elif dq is None:
```

File: src/robot_properties_solo/solo12wrapper.py [hunk 4]
```diff
@@ -132,3 +132,65 @@ def reset_to_initial_state(self) -> None:
...
+
+        self.pin_robot.forwardKinematics(q, dq)
+        self.pin_robot.computeJointJacobians(q)
+        self.pin_robot.framesForwardKinematics(q)
+        self.pin_robot.centroidalMomentum(q, dq)
+
+    def reset_to_initial_state(self) -> None:
+        """Reset robot state to the initial configuration (based on Solo12Config)."""
+        q0 = np.matrix(Solo12Config.initial_configuration).T
+        dq0 = np.matrix(Solo12Config.initial_velocity).T
```
Comment (nim65s) [inline src/robot_properties_solo/solo12wrapper.py]:
```suggestion
        q0 = np.array(Solo12Config.initial_configuration)
        dq0 = np.array(Solo12Config.initial_velocity)
```

File: src/robot_properties_solo/solo12wrapper.py [hunk 5]
```diff
@@ -132,3 +132,65 @@ def reset_to_initial_state(self) -> None:
...
+        - kinematics
+        - joint and frame jacobian
+        - centroidal momentum
+        Args:
+          q: Pinocchio generalized position vector.
+          dq: Pinocchio generalize velocity vector.
+        """
+        self.pin_robot.computeJointJacobians(q)
+        self.pin_robot.framesForwardKinematics(q)
+        self.pin_robot.centroidalMomentum(q, dq)
```
Comment (nim65s) [inline src/robot_properties_solo/solo12wrapper.py:196]:
```suggestion
        self.pin_robot.forwardKinematics(q, dq)
        self.pin_robot.computeJointJacobians(q)
        self.pin_robot.framesForwardKinematics(q)
        self.pin_robot.centroidalMomentum(q, dq)
```

**Review by nim65s — COMMENTED** [review]
Hi @swuuu,
Thanks for this contribution. I just made minor remarks.

**Comment by swuuu** [comment]
Hi @nim65s, I've committed your suggestions. Thanks!

**Pull request merged**